### DISCUSSION TITLE
Are extensions reinstalled every time a Codespace is restarted?

### DISCUSSION THREAD

**Discussion by alefragnani** [community/community#26853] [Codespaces]
<p>When you restart a Codespace (after its timeout/shutdown period), are the extensions <em>reinstalled</em>, or the VS Code server knows and reuses the previous session/workspace state?</p>
<p>I’m asking because when I return to a previous used Codespace (after a while) some installed extensions (like ESLint) appears to have a “fresh install” behavior, because it asks again if it should <em>“use <code>node_modules/eslint</code> for validation”</em></p>
<p>Thank you</p>

**Comment by sr229** [comment]
<p>It seems that the container environment is immutable, so it makes a new instance whenever you get out of it. The only persistent part is the volume mount used for it.</p>

**Comment by Chuxel** [comment, marked as answer]
<p>The entire container is persisted, so if you enter the same codespace, all your files will be there.  However, if you’re creating a new codespace each time, what you describe will happen.</p>

**Comment by alefragnani** [comment]
<p>That’s weird, but good to know :slight_smile: .</p>
<p>I had the feeling it wasn’t being persisted, but maybe I’ve recreated the workspace, and didn’t noticed. I just started using Codespaces and tried on a few different repos.</p>
<p>That’s great to know the container is persisted. Less work to do on my VS Code extensions :+1:</p>
<p>Thank you</p>
